viz_scripts/generic_metrics.ipynb

```diff
@@ -101,192 +102,493 @@
   },
   {
    "cell_type": "markdown",
-   "id": "modified-skiing",
+   "id": "6d44c87e",
    "metadata": {},
    "source": [
-    "## Generic Metrics"
+    "## Collect Data from Database for Sensed Metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55ec383b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "expanded_ct_sensed, file_suffix_sensed, quality_text_sensed, debug_df_sensed = scaffolding.load_viz_notebook_sensor_inference_data(year,\n",
+    "                                                                            month,\n",
+    "                                                                            program,\n",
+    "                                                                            include_test_users,\n",
+    "                                                                            sensed_algo_prefix)\n",
+    "\n",
+    "# Create a combined debug df from Generic and Sensed Metrics\n",
+    "merged_debug_df = debug_df.combine_first(debug_df_sensed)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "distributed-peace",
+   "id": "7f42f272",
    "metadata": {},
    "source": [
-    "### Distribution of Mode_confirm attribute"
+    "## Metrics for Specific Mode"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "tracked-serbia",
-   "metadata": {
-    "scrolled": false
-   },
+   "id": "657d7e18",
+   "metadata": {},
    "outputs": [],
    "source": [
-    "file_name='ntrips_mode_confirm%s' % file_suffix\n",
+    "data_eb = expanded_ct.query(f\"mode_confirm == '{mode_of_interest}'\") if \"mode_confirm\" in expanded_ct.columns else expanded_ct\n",
+    "quality_text_specific = scaffolding.get_quality_text(expanded_ct, data_eb, mode_of_interest, include_test_users)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1af1f02b",
+   "metadata": {},
+   "source": [
+    "## 100% Stacked Bar Charts"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e50959c1",
+   "metadata": {},
+   "source": [
+    "### 1. Represents Number of Trips"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aef741fe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def process_data_frame(df, df_col, trip_type):\n",
```

Comment:
why is this part of the same cell as `n_trips_total`? If it is a preprocessing step, it should be in a separate cell.